src/decorator-observable.js

```diff
@@ -36,10 +42,10 @@ export function observable(targetOrConfig: any, key: string, descriptor?: Proper
     delete descriptor.initializer;
 
     // add the getter and setter to the property descriptor.
-    descriptor.get = function() { return this[innerPropertyName]; };
+    descriptor.get = function() { return target[innerPropertyName]; };
     descriptor.set = function(newValue) {
-      let oldValue = this[innerPropertyName];
-      this[innerPropertyName] = newValue;
+      let oldValue = target[innerPropertyName];
+      target[innerPropertyName] = newValue;
```

Comment:
@Gheoan I don't think you can use `target` in this way. It creates a closure and is assigning the value to the prototype. I think if you write a test where you have two instances of an `@observable` decorated class, and you set the property on one instance, the second instance will have the same property value.